src/main/java/com/github/wovnio/wovnjava/Headers.java

```diff
@@ -238,7 +228,7 @@ public String locationWithLangCode(String location) {
         if (settings.urlPattern.equals("query") && location.contains("wovn=")) {
             return location;
         } else if (settings.urlPattern.equals("path")) {
-            Pattern p = Pattern.compile(settings.urlPatternReg);
+            Pattern p = this.patternHandler.getGetLangPattern();
```

Comment:
Seems like you could replace this with patternHandler.getLang.

Reply:
I think you're right. I hesitated to change this function for now, but I agree it's better to use `getLang` directly. That also lets me hide `getLangPattern`.